mutiny-core/src/lib.rs

```diff
@@ -2290,6 +2300,118 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    async fn cashu_mint_post<T: serde::de::DeserializeOwned>(
+        client: &reqwest::Client,
+        url: &Url,
+        body: Value,
+    ) -> Result<T, MutinyError> {
+        let res = client
+            .post(url.clone())
+            .header("Content-Type", "application/json")
+            .body(body.to_string())
+            .send()
+            .await
+            .map_err(|_| MutinyError::CashuMintError)?;
+        Self::parse_cashu_mint_response(res).await
+    }
+
+    async fn parse_cashu_mint_response<T: serde::de::DeserializeOwned>(
+        res: reqwest::Response,
+    ) -> Result<T, MutinyError> {
+        match res.status() {
+            StatusCode::OK => {
+                let response_text = res.text().await.map_err(|_| MutinyError::CashuMintError)?;
+                match serde_json::from_str::<T>(&response_text) {
+                    Ok(data) => Ok(data),
+                    Err(_) => Err(MutinyError::CashuMintError),
+                }
+            }
+            _ => {
+                let txt = res.text().await.map_err(|_| MutinyError::CashuMintError)?;
+                let data = serde_json::from_str::<CashuErrorResponse>(&txt)
+                    .map_err(|_| MutinyError::CashuMintError)?;
+
+                match data.code {
+                    // error code in nutshell for tokens that have been spent
+                    11001 => Err(MutinyError::TokenAlreadySpent),
+                    _ => Err(MutinyError::CashuMintError),
+                }
+            }
+        }
+    }
+
+    /// Calls upon a Cashu mint and redeems/melts the token.
+    pub async fn melt_cashu_token(
+        &self,
+        token_v3: TokenV3,
+    ) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let client = reqwest::Client::new();
+        let mut invoices: Vec<MutinyInvoice> = Vec::with_capacity(token_v3.tokens.len());
+
+        for token in token_v3.tokens {
+            let mint_url = match token.mint {
+                Some(url) => url,
+                None => return Err(MutinyError::EmptyMintURLError),
+            };
+
+            let total_proofs_amount = token.proofs.total_amount();
+            // create invoice for 1% less than proofs amount
+            let mut invoice_amount = total_proofs_amount as f64 * (1.0 - 1.0 / 100.0);
+            let mut mutiny_invoice: MutinyInvoice;
+            let mut mutiny_invoice_str: Bolt11Invoice;
+            let mut melt_quote_res: PostMeltQuoteBolt11Response;
```

Comment:
since you aren't using these outside the loop there's no need to initialize them out here with a `mut`.

Reply:
yeah I'm using them outside the loop

```
if post_melt_bolt11_response.paid {
    mutiny_invoice = self.get_invoice(&mutiny_invoice_str).await?;
    invoices.push(mutiny_invoice);
}
```